src/core/ops_internal.cpp

```diff
@@ -610,4 +610,88 @@ bool operator==(const SHTypeInfo &a, const SHTypeInfo &b) {
   default:
     return true;
   }
-}
\ No newline at end of file
+}
+
+ALWAYS_INLINE inline bool _almostEqual(const float a, const float b, const double e) { return __builtin_fabsf(a - b) <= e; }
+ALWAYS_INLINE inline bool _almostEqual(const double a, const double b, const double e) { return __builtin_fabs(a - b) <= e; }
+ALWAYS_INLINE inline bool _almostEqual(const int32_t a, const int32_t b, const double e) { return __builtin_abs(a - b) <= e; }
+ALWAYS_INLINE inline bool _almostEqual(const int64_t a, const int64_t b, const double e) { return __builtin_llabs(a - b) <= e; }
+
+bool _almostEqual(const SHVar &lhs, const SHVar &rhs, double e) {
+  if (lhs.valueType != rhs.valueType) {
+    return false;
+  }
+
+  switch (lhs.valueType) {
+  case SHType::Float:
+    return _almostEqual(lhs.payload.floatValue, rhs.payload.floatValue, e);
+  case SHType::Float2:
+    return (_almostEqual(lhs.payload.float2Value[0], rhs.payload.float2Value[0], e) &&
+            _almostEqual(lhs.payload.float2Value[1], rhs.payload.float2Value[1], e));
+  case SHType::Float3:
+    return (_almostEqual(lhs.payload.float3Value[0], rhs.payload.float3Value[0], e) &&
+            _almostEqual(lhs.payload.float3Value[1], rhs.payload.float3Value[1], e) &&
+            _almostEqual(lhs.payload.float3Value[2], rhs.payload.float3Value[2], e));
+  case SHType::Float4:
+    return (_almostEqual(lhs.payload.float4Value[0], rhs.payload.float4Value[0], e) &&
+            _almostEqual(lhs.payload.float4Value[1], rhs.payload.float4Value[1], e) &&
+            _almostEqual(lhs.payload.float4Value[2], rhs.payload.float4Value[2], e) &&
+            _almostEqual(lhs.payload.float4Value[3], rhs.payload.float4Value[3], e));
```

Comment:
Vectors should use vector math

Reply:
What kind of vector math do you want to use here?

Unrolling the loops is what the compiler will do anyway: https://godbolt.org/z/8KTxGncfe

Reply:
We do have actual comparisons operators with vectors implemented for Var. Please check again. 👍

Reply:
Exactly in ops.hpp btw

Reply:
Err I see what you mean. Why this happened? @guusw was you or me? We should use vectors just like `==` compiler is not smart enough. 
Maybe you can refactor that too @Kryptos-FR actually. It's the whole point for us using gcc/clang vectors...

Reply:
But you know what, we can avoid for now if it's annoying. Didn't realize it's so convoluted.